pkgs/development/ocaml-modules/awa/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, buildDunePackage, fetchurl
+, ppx_sexp_conv, ppx_cstruct
+, mirage-crypto, mirage-crypto-rng, mirage-crypto-pk
+, x509, cstruct, cstruct-unix, cstruct-sexp, sexplib
+, rresult, mtime, logs, fmt, cmdliner, base64, hacl_x25519
+, zarith
+}:
+
+buildDunePackage rec {
+  pname = "awa";
+  version = "0.0.1";
+
+  minimumOCamlVersion = "4.07";
+  useDune2 = true;
+
+  src = fetchurl {
+    url = "https://github.com/mirage/awa-ssh/releases/download/v${version}/awa-v${version}.tbz";
+    sha256 = "c1d604645517b191184a5800f5c48a6a9a3e5a2fce4a0e2ecfeee74586301400";
+  };
+
+  nativeBuildInputs = [ ppx_sexp_conv ppx_cstruct ];
+
+  propagatedBuildInputs = [
+    mirage-crypto mirage-crypto-rng mirage-crypto-pk x509
+    cstruct cstruct-unix cstruct-sexp sexplib rresult mtime
+    logs fmt cmdliner base64 hacl_x25519 zarith
```

Comment:
Libraries `cstruct-unix` and `cmdliner` are build-time only (better not to propagate them).

Reply:
Good catch, `fmt` can also be moved to `checkInputs`.